### PR TITLE
docs: add upstream sync and setup shortcuts

### PR DESCRIPTION
**Pull request by jonathan-irvin**
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Documentation Update

## Description

This adds two commands:
* `npm run upstream:connect` Adds Dev.to remote to a local repo.  In most cases, this will be a fork.
* `npm run upstream:sync` Does a checkout on master, then does a pull from latest commit on dev.to master and merges with local branch

## Related Tickets & Documents
Related to #740 and #742

## Added to documentation?
  - [ ] docs.dev.to
  - [x] readme
  - [ ] no documentation needed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,44 @@ When in doubt, ask a [core team member](#core-team)! You can mention us in any i
...
+Inside your dev.to directory, add a remote to the official dev.to repo:
+```
+$ git remote add upstream https://github.com/thepracticaldev/dev.to.git
+```
+
+Optionally, you can run two commands to setup and sync your fork:
+  * `npm run upstream:setup` Will perform the above command
+  * `npm run upstream:sync` Will checkout the master branch and pull the master branch from Dev.to into your local master branch.
+
+#### Rebasing from Upstream
```
Comment (rhymes) [inline README.md]: What about people who don't want to rebase?
GitHub's guide might also be worth linking: https://help.github.com/articles/syncing-a-fork/

Comment (jonathan-irvin) [reply README.md]: `npm run upstream:sync` won't be a rebase, it would be a merge.  I've included the documentation from another PR to supplement.

**Comment by jonathan-irvin** [comment]
@maestromac Great points.  I added the Npm scripts (yes, while harmless) is pretty easy to understand.  That's the point, right?

That way in your command line, after you've cloned your fork, the end-user would only need to run two commands `npm run upstream:setup` and then `npm run upstream:sync` going forward.  I agree the commands could be shortened to something like `npm run link` and `npm run sync`. It rhymes!

> The Maintaining Your Fork section is a nice addition but it makes more sense for it to be a in the docs folder.

As for this, the docs folder is great for those of us who know what the heck we're doing, but I agree with @nathansebhastian in originally having it live in the main README **only** because it's the ideal one-stop-shop for early contributors.  It's a nice little "oh, by the way" this is how you can work with forks.

GitHub is great with the forking concept, but it doesn't readily tell you how to sync your fork with the upstream without doing a multitude of PRs from dev.to to the fork.

Supplemental Docs:
* [Syncing a Fork](https://help.github.com/articles/syncing-a-fork/)
* [This Gist that says the same things](https://gist.github.com/CristinaSolana/1885435)

**Comment by maestromac** [comment]
My apology for the delay and once again, thank you both for this PR.

Keeping a forked branch up-to-date and dealing with bizarre merge or rebase conflict is something every developer will encounter, learn, and overcome. A few lines of code and script to help out beginner is noble but I believe such process, whether to name the remote "upstream" or when to rebase vs merge,  should be dealt with at a developer's discretion and not something [dev.to](http://dev.to) dictates.

 @nathansebhastian I'm going to reopen your PR to ticker with and merge it.

Closing

**Pull request closed without merging**